custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -0,0 +1,194 @@
+        dcpulmunory # Pending
+        dotname
+        dotdesignation
+        dotmob
+        dotpType
+        dotcenter # Pending
+        regBy # Used date_of_diagnosis instead of date_of_registration
+        IP_From # Pending
+        Local_ID
+        tcode # Pending
+        page
+        poccupation
+        dotphi
+        Ptype
+        pcategory # Pending
+        Source
+        """
+        # properties skipped with example values
+        # "plandline": "N/A",
+        # "clandline": "N/A",
+        # "dotlandline": "N/A",
+        # "cvisitedby": "N/A",
+        # "cvisitedDate": "N/A",
+        # "dotcenter": "N/A",
+        # "dotmoname": "N/A",
+        # "mosign": "N/A",
+        # "mosigndonedate": "N/A",
+        # "atbtreatment": "N/A",
+        # "atbduration": ,
+        # "atbsource": "G",
+        # "atbregimen": "Manish-Regim",
+        # "atbyr": "2015",
+        # "dcpulmunory": "P",
+        # "IP_Address": "mk-ip-address",
+        # "IP_From": "127.0.0.1",
+        person_case = get_person_case_from_episode(repeat_record.domain, episode_case.get_id)
+
+        episode_case_properties = episode_case.case_properties()
+        person_case_properties = person_case.case_properties()
+
+        properties_dict = {
+            "regBy": self.repeater.username,
+            "Local_ID": person_case.get_id,
+            "Source": ENIKSHAY_ID
+        }
+        properties_dict.update(_get_person_case_properties(person_case_properties))
+        properties_dict.update(_get_episode_case_properties(episode_case_properties))
+        return json.dumps(properties_dict)
+
+    def handle_success(self, response, payload_doc, repeat_record):
+        # A success would be getting a nikshay_id for the patient
+        # without it this would actually be a failure
+        try:
+            nikshay_id = response.json()['Results']['Fieldvalue']
+            person_case_id = get_person_case_from_episode(payload_doc.domain, payload_doc.case_id).get_id
+            _update_person_case(
+                payload_doc.domain,
+                person_case_id,
+                {
+                    "nikshay_registered": "true",
+                    "nikshay_id": nikshay_id
+                }
+            )
+        except (JSONDecodeError, KeyError):
+            self.handle_failure(response, payload_doc, repeat_record)
+
+    def handle_failure(self, response, payload_doc, repeat_record):
+        try:
+            error_message = response.json().get('Results', response.content)
+        except (JSONDecodeError, KeyError):
+            error_message = response.content
+        save_error_message(response.status_code, error_message)
+
+
+def _get_person_case_properties(person_case_properties):
+    state_choice = person_case_properties.get('current_address_state_choice', None)
+    district_choice = person_case_properties.get('current_address_district_choice', None)
+
+    if state_choice:
+        state_location = (SQLLocation.objects.
```

Comment:
I think you'll want to handle the case where this location is not found (i.e. just fall back to a default value). This will throw an exception if `.get` returns nothing.